### PR TITLE
fix(filters): add border type to median blur

### LINKED ISSUE #4670: [Bug]: median_blur zero-pads and has no border_type, so it returns 0 in the corners of a constant image
### What happened?

`median_blur` zero-pads the image and offers no `border_type`, so it returns 0 in the corners of a constant image. Every other filter in `kornia.filters` takes `border_type` and defaults to `"reflect"`.

On a uniform 0.8 image, 11x11:

| filter | corner | centre |
| --- | --- | --- |
| `median_blur(img, (5,5))` | **0.000** | 0.800 |
| `box_blur(img, (5,5))` | 0.800 | 0.800 |
| `gaussian_blur2d(img, (5,5), (1.,1.))` | 0.800 | 0.800 |

A blur that changes a constant image is wrong independently of which padding one prefers.

It is the corners only, not the whole border, because a median needs the padding to be a majority of the window. The affected region is a triangle at each corner that grows with the kernel: 4 of 121 pixels at 3x3, 12 at 5x5, 20 at 7x7.

Both code paths do this — `F.conv2d(..., padding=padding)` on the general path and `F.pad(input, (radius,)*4)` in `_median_blur_network`, whose default mode is `"constant"` — so they agree with each other and disagree with the rest of the module.

Checked against `scipy.ndimage.median_filter` on random input: kornia matches `mode="constant"` exactly and no other mode, while the interior matches all of them. `gaussian_blur2d` and `box_blur` match scipy to 3e-16 on all four of their border types, so this is specific to `median_blur`.

Two ways to resolve it, and the choice is yours:

1. Add `border_type: str = "reflect"` and pass it through, matching the siblings. Output changes for existing users in those corner triangles.
2. Add `border_type: str = "constant"` and document the default. Nothing changes for existing users, and the surprise becomes discoverable.

I would pick 1, because the current output is not a median of any window the user supplied. But 2 is the compatible one and I would rather you decided.

Happy to send the PR for whichever you prefer.

### Minimal reproduction

```python
import torch
from kornia.filters import median_blur, box_blur

img = torch.full((1, 1, 11, 11), 0.8)

print(median_blur(img, (5, 5))[0, 0, 0, 0])  # tensor(0.)
print(box_blur(img, (5, 5))[0, 0, 0, 0])     # tensor(0.8000)

out = median_blur(img, (5, 5))
print((out != 0.8).sum().item(), "of", out.numel(), "pixels changed")  # 12 of 121
```

### Environment

- Kornia version: 0.9.0rc1 (main at e1b79c21)
- PyTorch version: 2.13.0
- Python version: 3.14.0
- OS / device (CPU, CUDA, ROCm, MPS): macOS 26.5.2 arm64, CPU

`kornia 0.9.0rc1 | torch 2.13.0 | python 3.14.0 | OS macOS-26.5.2-arm64-arm-64bit-Mach-O | CUDA/ROCm None (available=False, GPU=none) | MPS=True`

### Anything else?

The suite cannot catch this, in three separate ways, which is probably why it has lasted:

- `test_kernel_3x3` asserts only interior pixels, `actual[0, 0, 2, 2]` and `actual[0, 1, 1, 1]`. No test reads a corner.
- `test_selection_nonfinite` states the padding as the contract in a comment: *"Finite windows contain only ones and zero padding, so their median is one exactly when a majority of the window lies inside the image."*
- The parity tests build `expected` with `torch.nn.functional.conv2d(inp, weights, padding=kernel_size // 2)`, which is the same zero padding the implementation uses, so the expectation cannot represent any other border behaviour.

Same shape as #3580, where `median_blur` was the one filter that did not handle `B=0`.

Written with Claude Code (Claude Opus 5) on my account. The scipy comparison and the pixel counts above are its measurements, not estimates.

### PR DESCRIPTION
**Pull request by Kakarot2812**
## What does this pull request do?

Adds a `border_type` argument to `median_blur`, defaulting to `"reflect"`.

The padding mode is now applied consistently to both the optimized median-blur network path and the convolution path. Supported modes are `constant`, `reflect`, `replicate`, and `circular`.

`MedianBlur` also accepts and forwards `border_type`.

Tests were updated to cover the new padding behavior while preserving the existing `constant` behavior when explicitly requested.

## Related issue or discussion

Fixes #4670

## What did you check?

- `tests/filters/test_median.py` — 106 passed
- `tests/filters` — 1210 passed, 10 skipped
- `tests/augmentation/test_conventions_intensity_ops.py` — 3 passed
- `tests/augmentation/test_augmentation.py -k "test_feature_median_blur"` — 1 passed
- `tests/augmentation` — 3468 passed, 3405 skipped, 30 xfailed, 5 xpassed
- `pixi run lint` — passed
- `git diff --check` — passed

## How did AI help?

AI was used for development guidance, code review, and test/debugging suggestions. The implementation was manually reviewed, and the changes and test results were verified locally before opening this pull request.

### CODE REVIEW

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
## [P2] Do not silently break the default border contract

kornia/filters/median.py:97 changes the existing default from constant padding to reflect. That changes every caller which omits border_type and makes previously valid thin inputs fail; the new convention test explicitly records RandomMedianBlur on a 1x8 input raising under the new default. The fixed fragment merely says the default is now reflect and does not disclose the old output contract or the new runtime failure. Preserve constant as the default while adding the new option, or, if this break is intentional, classify and document it as breaking with the old-to-new behavior and reflect-padding size constraint. The cancelled float16 job and failed collector also need a successful rerun.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5).

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
[P2] Apply circular padding to the nonfinite mask — `kornia/filters/median.py:86`

The optimized path pads sample values with `border_type` but still builds its nonfinite mask by max-pooling the unpadded input with implicit constant padding. With circular padding, an Inf at a corner participates in windows on the opposite edges. For a `(1,1,7,9)` tensor of ones with `x[...,0,0] = inf`, `median_blur(x, 3, "circular")` on CPU/no-grad marks NaN only near the source corner, while the convolution/autograd path also marks `[0,8]`, `[1,8]`, `[6,0]`, `[6,1]`, and `[6,8]`. The output therefore depends on dispatch/autograd state. Pad the nonfinite mask with the selected border mode before pooling and add a corner/edge regression; the current test places the invalid value at the center, where all modes coincide.

The previous default-padding compatibility finding is addressed. The 113 existing median tests pass but do not cover this edge case.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).
